Cartfile

```diff
@@ -1,4 +1,4 @@
-binary "https://www.mapbox.com/ios-sdk/Mapbox-iOS-SDK.json" ~> 4.2
+binary "https://www.mapbox.com/ios-sdk/Mapbox-iOS-SDK.json" ~> 4.3
```

Comment:
If semver patch releases of the maps SDK are potentially incompatible (due to close coupling, private API usage, etc.), does it make sense to tighten the pin here to `~> 4.3.0`?

Reply:
With this PR, private API usage is down to a fairly benign minimum – just a forward declaration of `-mapViewDidFinishRenderingFrameFullyRendered:`. The broader root cause in #1607 is that Swift is strict about subclasses having naming collisions with superclasses. It makes you wonder whether we should be adding class prefixes to NavigationMapView properties that are likely to collide with MGLMapView properties in the future. 🤔

I’d rather not specify a patch release because, on balance, fire drills like this are rare. We’d generally want developers to be able to adopt new map SDK versions as they come out; at the same time, we don’t want to prevent developers from using a previous minor version of the map SDK if they require it for some reason. It’s a tough tradeoff, but I think we can avoid future problems just by being more aware of potential pitfalls when adding to MGLMapView or NavigationMapView.